File: 백준/Bronze/25304. 영수증/영수증.py
import sys
input = sys.stdin.readline

X = int(input())
N = int(input())

i = 0
total = 0
while i < N:
    a, b = map(int, input().split())
    total += a * b
    i += 1

if X == total:
    print("Yes")
else:
    print("No")

File: 백준/Bronze/25304. 영수증/영수증.js
let fs = require('fs')
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

X = input[0]
N = input[1]

let total = 0
for (let i = 2; i < input.length; i++) {
    line = input[i].split(" ")
    total += parseInt(line[0]) * parseInt(line[1])
}

if (parseInt(X) === total) {
    console.log("Yes")
} else {
    console.log("No")
}